Dockerfile

```diff
@@ -1,11 +1,14 @@
 ARG BASE_IMAGE="debian"
 ARG BASE_IMAGE_TAG="buster"
-FROM ${BASE_IMAGE}:${BASE_IMAGE_TAG}
+FROM --platform=$TARGETPLATFORM ${BASE_IMAGE}:${BASE_IMAGE_TAG}
+
+ARG TARGETPLATFORM
+
+ARG ARCH=${TARGETPLATFORM}
+
+
 
 # Arguments to instantiate as variables
-ARG BASE_IMAGE
-ARG BASE_IMAGE_TAG
-ARG ARCH="amd64"
```

Comment:
These lines make sure these environment variables will still be available after "FROM". They shall not be removed to make sure this information can be accessed later on during runtime (e.g. by dockerinfo module)

Reply:
These are no ENV variables, these are ARG, which are available only during Build process, but not part of the image.
And because i haven't found any location where these ARG are used, i removed them to clean up orphans.


> ARG variables are not persisted into the built image as ENV variables are

Reply:
these ARGs are there so people can use them when building a local image themselves.

Reply:
you have to repeat BASE_IMAGE and BASE_IMAGE_TAG after FROM, otherwise they will not be available for other commands anymore (e.g. if used by somebody adding some customizations to the build process)

Reply:
If someone want's to modify the build process ( dockerfile ) he can add it where he needs it i think.
I removed it, because currently it's nowhere used, but it woud break caching as soon as it is defined with a different value.